File: exercicios/kauan/models/lista01/Livro.cpp
#include "models/lista01/Livro.h"

#include "utils/ValorString.h"


Livro::Livro(ValorString titulo, ValorString autor, int paginas):titulo(titulo), autor(autor), paginas(paginas){}

ValorString Livro::obterResumo(){
            return ValorString("O livro " + this->titulo.str() + " escrito pela autora " + this->autor.str() + " possui " + ValorString::intParaString(this->paginas).str() + " páginas");
        }



void Livro::alterarNumeroPaginas(int paginas){
    this->paginas = paginas;
}

File: exercicios/aula-01/Gabriele Cardoso. Aula1/Livro.java
public class Livro {
    String titulo;
    String autor;
    int paginas;
}

public class TesteLivro {
    public static void main(String[] args) {

        Livro meuLivro = new Livro();

        meuLivro.titulo = "Java para Iniciantes";
        meuLivro.autor = "João Pereira";
        meuLivro.paginas = 180;

        System.out.println("=== Informações do Livro ===");
        System.out.println(meuLivro.titulo + ", escrito por "
                + meuLivro.autor + ", possui "
                + meuLivro.paginas + " páginas.");

        // Atualizando número de páginas
        meuLivro.paginas = 220;

        System.out.println("\n=== Após atualização ===");
        System.out.println(meuLivro.titulo + ", escrito por "
                + meuLivro.autor + ", agora possui "
                + meuLivro.paginas + " páginas.");
    }
}